starlark/testdata/string.star

```diff
@@ -174,6 +174,8 @@ assert.eq("a{x}b{y}c{}".format(1, x=2, y=3), "a2b3c1")
 assert.fails(lambda: "a{z}b".format(x=1), "keyword z not found")
 assert.fails(lambda: "{-1}".format(1), "keyword -1 not found")
 assert.fails(lambda: "{-0}".format(1), "keyword -0 not found")
+assert.fails(lambda: "{+0}".format(1), "keyword \\+0 not found")
+assert.fails(lambda: "{+1}".format(1), "keyword \\+1 not found") # starlark-go/issues/114
```

Comment:
```python
assert.eq("{0000000000001}".format(0,1), "1")
```

(This matches Python 2/3 behavior, and makes explicit to future readers of the code that this behavior is intended.)

Reply:
Good point. This much better than a length check.

Reply:
I added this one too:
assert.eq("{012}".format(*range(100)), "12") # decimal, despite leading zeros